modules/pricer/src/main/java/com/opengamma/strata/pricer/fx/SmileDeltaTermStructureParametersStrikeInterpolation.java

```diff
@@ -206,297 +272,812 @@ public SmileDeltaTermStructureParametersStrikeInterpolation(
    * {@code volatility} should be {@code n * (2 * m + 1)}, where {@code n} is the length of {@code timeToExpiry}
    * and {@code m} is the length of {@code delta}.
    * 
-   * @param name  the name of smile parameter term structure 
-   * @param timeToExpiry  the time to expiry of each volatility smile
-   * @param delta  the delta at which the volatilities are given 
-   * @param volatility  the volatilities at each delta 
-   * @param strikeInterpolator  the interpolator used in the strike dimension 
+   * @param name  the name
+   * @param timeToExpiry  the expiry times of individual volatility smiles
+   * @param delta  the delta values
+   * @param volatility  the volatilities
+   * @param timeLeftExtrapolator  left extrapolator used in the time dimension
+   * @param timeInterpolator  interpolator used in the time dimension
+   * @param timeRightExtrapolator  right extrapolator used in the time dimension
+   * @param strikeLeftExtrapolator  left extrapolator used in the strike dimension
+   * @param strikeInterpolator  interpolator used in the strike dimension
+   * @param strikeRightExtrapolator  right extrapolator used in the strike dimension
+   * @return the instance
    */
-  public SmileDeltaTermStructureParametersStrikeInterpolation(
+  public static SmileDeltaTermStructureParametersStrikeInterpolation of(
       String name,
-      double[] timeToExpiry,
-      double[] delta,
-      double[][] volatility,
-      Interpolator1D strikeInterpolator) {
-    super(name, timeToExpiry, delta, volatility);
-    _strikeInterpolator = ArgChecker.notNull(strikeInterpolator, "strikeInterpolator");
+      DoubleArray timeToExpiry,
+      DoubleArray delta,
+      DoubleMatrix volatility,
+      CurveExtrapolator timeLeftExtrapolator,
+      CurveInterpolator timeInterpolator,
+      CurveExtrapolator timeRightExtrapolator,
+      CurveExtrapolator strikeLeftExtrapolator,
+      CurveInterpolator strikeInterpolator,
+      CurveExtrapolator strikeRightExtrapolator) {
+
+    ArgChecker.notNull(timeToExpiry, "time to expiry");
+    ArgChecker.notNull(delta, "delta");
+    ArgChecker.notNull(volatility, "volatility");
+    ArgChecker.isTrue(delta.size() > 0,
+        "Need more than one volatility value to perform strike interpolation");
+    int nbExp = timeToExpiry.size();
+    ArgChecker.isTrue(volatility.rowCount() == nbExp,
+        "Volatility array length {} should be equal to the number of expiries {}", volatility.rowCount(), nbExp);
+    ArgChecker.isTrue(volatility.columnCount() == 2 * delta.size() + 1,
+        "Volatility array {} should be equal to (2 * number of deltas) + 1, have {}",
+        volatility.columnCount(), 2 * delta.size() + 1);
+    SmileDeltaParameters[] vt = new SmileDeltaParameters[nbExp];
+    for (int loopexp = 0; loopexp < nbExp; loopexp++) {
+      vt[loopexp] = SmileDeltaParameters.of(
+          timeToExpiry.get(loopexp), delta, DoubleArray.copyOf(volatility.rowArray(loopexp)));
```

Comment:
Should use `volatility.row(loopexp)` to avoid copy. Same pattern appears in other methods in this class